web/src/engine/websites/Kakaopage.ts

```diff
@@ -98,23 +100,16 @@ export default class extends DecoratableMangaScraper {
         `;
 
         do {
-            const request = new Request(new URL('/graphql', this.URI).href, {
-                headers: {
-                    Referer: this.URI.href,
-                    Origin: this.URI.href
-                }
-            });
-
             const vars: JSONObject = {
                 seriesId: parseInt(manga.Identifier),
                 boughtOnly: false,
                 sortType: 'asc',
                 after: nextCursor
             };
 
-            const data = await FetchGraphQL<APIChapters>(request, 'contentHomeProductList', gql, vars);
-            const chapters = data.contentHomeProductList.edges.map(chapter => new Chapter(this, manga, chapter.node.single.productId.toString(), chapter.node.single.title.replace(manga.Title, '').trim()));
-            nextCursor = data.contentHomeProductList.pageInfo.hasNextPage ? data.contentHomeProductList.pageInfo.endCursor : null;
+            const { contentHomeProductList: { edges, pageInfo: { hasNextPage, endCursor } } } = await FetchGraphQL<APIChapters>(this.CreateRequest(), 'contentHomeProductList', gql, vars);
+            const chapters = edges.map(chapter => new Chapter(this, manga, chapter.node.single.productId.toString(), chapter.node.single.title.replace(manga.Title, '').trim()));
+            nextCursor = hasNextPage ? endCursor : undefined;
```

Comment:
Since `undefined` is a valid value for `nextCursor`, consider to use `hasNextPage` or a `run` variable as condition to break the loop to avoid confusion